### PR TITLE
debug log number of unknown wallet records on load

### PR DESCRIPTION
**Pull request by instagibbs**
This would have saved me some time during wallet debugging, with minimal logging clutter.

### CODE REVIEW

**Comment by Empact** [comment]
How about logging the unknown record `strType`s themselves, rather than the number? Would provide additional info and avoid the need for storing the state.

**Comment by instagibbs** [comment]
I've done a spammier approach on my own branches, but I suppose I could throw an `std::set` at the problem if people aren't against the idea.

File: src/wallet/walletdb.cpp [hunk 1]
```diff
@@ -597,6 +600,7 @@ DBErrors CWalletDB::LoadWallet(CWallet* pwallet)
 
     LogPrintf("Keys: %u plaintext, %u encrypted, %u w/ metadata, %u total\n",
            wss.nKeys, wss.nCKeys, wss.nKeyMeta, wss.nKeys + wss.nCKeys);
+    LogPrintf("Unknown wallet records: %u\n", wss.m_unknown_records);
```
Comment (practicalswift) [inline src/wallet/walletdb.cpp]: To reduce the amount of logging I suggest printing this only if `wss.m_unknown_records != 0`.

Alternatively add `wss.m_unknown_records` printing to the already existing log entry (`Keys: …`) above.

Comment (laanwj) [reply src/wallet/walletdb.cpp]: > To reduce the amount of logging I suggest printing this only if wss.m_unknown_records != 0.

Or print it on the same line as the `Keys:`. In that case it doesn't matter to log it every time.

**Comment by laanwj** [comment]
ConceptACK

File: src/wallet/walletdb.cpp (hunk 1 above)
Comment (practicalswift) [reply src/wallet/walletdb.cpp]: @laanwj Agreed! That was suggestion 2 :-) My vote is on suggestion 2.

Comment (promag) [reply src/wallet/walletdb.cpp]: 👍 @practicalswift

**Comment by promag** [comment]
Concept ACK.

@instagibbs can you detail the purpose of this counter?

**Comment by laanwj** [comment]
If you want to log spammier stuff, put it in the "wallet" category. A total
should suffice in the default case.

File: src/wallet/walletdb.cpp (hunk 1 above)
Comment (instagibbs) [reply src/wallet/walletdb.cpp]: I split it out because the print line is "Keys:", but I suppose this could have meant "Key/Value" versus "crypto keys"?

Comment (instagibbs) [reply src/wallet/walletdb.cpp]: It's counting only the crypto keys as "total", not metadata which I think led me to believe this.

If I add it to the above line, I suppose I'll ommit it from "total"?

**Comment by practicalswift** [comment]
Agree with @laanwj after thinking about it.

Put this in the wallet category. Logging to the default log level should be reserved for things that are likely to be of interest for the end-user.

**Comment by promag** [comment]
@instagibbs I mean in what way is this counter useful?

**Comment by instagibbs** [comment]
@promag lets you know if there are "num" amount of records your wallet doesn't recognize. In my case it would have told me exactly what was going on(type typo).

File: src/wallet/walletdb.cpp (hunk 1 above)
Comment (laanwj) [reply src/wallet/walletdb.cpp]: Indeed, I didn't mean to include it in keys but have it seperately e.g.
```
Keys: %u plaintext, %u encrypted, %u w/ metadata, %u total. Unknown wallet records: %u.
```

**Comment by instagibbs** [comment]
put on single log entry as requested by @laanwj

**Comment by laanwj** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/12888/commits/6b25bae87c03d54d9624d4cce3f76f0033df7459

**Comment by practicalswift** [comment]
utACK 6b25bae87c03d54d9624d4cce3f76f0033df7459

**Comment by instagibbs** [comment]
one test timed out, kicked it. Maybe it'll be faster after the unit test improvements...

**Comment by laanwj** [comment]
Noticed while testing: This should probably ignore `bestblock` and `bestblock_nomerkle` as it does write those.

**Comment by instagibbs** [comment]
updated to filter `bestblock` and `bestblock_nomerkle` entries.

**Comment by laanwj** [comment]
tested ack 6979ddc4f2cada2b411eccc41cc5fed1c8c6ba44

**Comment by laanwj** [comment]
Looks like we forgot to account for `minversion` here.

**Pull request merged**